homeassistant/components/homematicip_cloud/__init__.py

```diff
@@ -289,6 +295,28 @@
         schema=SCHEMA_DUMP_HAP_CONFIG,
     )
 
+    async def _async_reset_energy_counter(service):
+        """Service to reset the energy counter."""
+        entity_id_list = service.data[ATTR_ENTITY_ID]
+
+        for hap in hass.data[DOMAIN].values():
+            if entity_id_list != "all":
+                for entity_id in entity_id_list:
+                    device = hap.hmip_device_by_entity_id.get(entity_id)
+                    if device and isinstance(device, AsyncSwitchMeasuring):
+                        await device.reset_energy_counter()
+            else:
+                for device in hap.home.devices:
+                    if isinstance(device, AsyncSwitchMeasuring):
+                        await device.reset_energy_counter()
+
+    hass.services.async_register(
```

Comment:
We should register this as an admin service. Use the helper in `helpers/service.py`

Reply:
done